core/domain/learner_progress_services_test.py

```diff
@@ -481,6 +507,89 @@ def test_get_all_completed_exp_ids(self):
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_exploration(system_user, self.EXP_ID_3)
+        private_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_3)
+        self.assertEqual(
+            private_exploration.status, constants.ACTIVITY_STATUS_PRIVATE)
+
+        # Call get_activity_progress to get filtered progress.
+        user_activity = learner_progress_services.get_activity_progress(
+            self.user_id)
+        all_filtered_summaries = user_activity[0]
+        completed_exp_summaries = (
+            all_filtered_summaries.completed_exp_summaries)
+
+        # Test that completed exp summaries don't include private exploration.
+        # Ensure that completed_exp_summaries[0] matches EXP_ID_0.
+        self.assertEqual(
+            completed_exp_summaries[0].id, '0_en_arch_bridges_in_england')
+        # Ensure that completed_exp_summaries[1] matches EXP_ID_1.
+        self.assertEqual(
+            completed_exp_summaries[1].id, '1_fi_arch_sillat_suomi')
+        self.assertEqual(len(completed_exp_summaries), 2)
+
+    def test_publishing_private_exploration_is_filtered_as_complete(self):
+        # Add explorations to the completed list.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_0)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_1)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_3)
+
+        # Unpublish EXP_ID_3 to change status to ACTIVITY_STATUS_PRIVATE.
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_exploration(system_user, self.EXP_ID_3)
+        private_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_3)
+        self.assertEqual(
+            private_exploration.status, constants.ACTIVITY_STATUS_PRIVATE)
+
+        # Publish EXP_ID_3 to change status back to ACTIVITY_STATUS_PUBLIC.
+        self.publish_exploration(self.owner_id, self.EXP_ID_3)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_3)
+        public_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_3)
+        self.assertEqual(
+            public_exploration.status, constants.ACTIVITY_STATUS_PUBLIC)
+
+        # Call get_activity_progress to get filtered progress.
+        user_activity = learner_progress_services.get_activity_progress(
+            self.user_id)
+        all_filtered_summaries = user_activity[0]
+        completed_exp_summaries = (
+            all_filtered_summaries.completed_exp_summaries)
+
+        # Test that completed exp summaries includes all 3 of the explorations.
+        self.assertEqual(len(completed_exp_summaries), 3)
```

Comment:
Assert the exploration ids (similar to line 539)

Reply:
Done :)